python/ray/serve/api.py

```diff
@@ -718,3 +719,32 @@ def my_deployment_function(request):
     """
     _request_context = ray.serve.context._serve_request_context.get()
     return _request_context.multiplexed_model_id
+
+
+@PublicAPI(stability="alpha")
+def status() -> List[Dict]:
+    """Get statuses of all active applications on the cluster.
+
+    Each dictionary in the list matches the format of ServeStatusSchema,
+    which includes the application status, all deployment statuses,
+    status messages, etc. If Serve hasn't been started on the cluster
+    yet, this returns an empty list.
+
+    .. code-block:: python
+
+            @serve.deployment(num_replicas=2)
+            class MyDeployment:
+                pass
+
+            app = MyDeployment.bind()
+            serve.run(app)
+            status = serve.status()
+            assert status[0]["app_status"]["status"] == "RUNNING"
+    """
+
+    try:
+        client = get_global_client()
+        details = ServeInstanceDetails(**client.get_serve_details())
+        return [app.get_status_dict() for app in details.applications.values()]
+    except RayServeException:
+        return []
```

Comment:
this is very sketchy. can we look for a more explicit signal?

Reply:
Do you mean returning an empty list when the controller isn't running? We could return `None` instead, not sure what is preferred. I wanted to mirror the behavior of `serve status`, and not throw an error when the controller hasn't started yet.

Reply:
Gotcha. I've made changes to the `get_global_client` function to take in an internal flag that makes it explicitly return None if there is no controller running, then use that to decide whether to return an empty list.